src/components/share-buttons/share-buttons.component.ts

```diff
@@ -38,6 +38,9 @@ export class ShareButtonsComponent implements OnInit {
     /** Indicates weather default style is applied to the buttons */
     @Input() defaultStyle: boolean = true;
 
+    /** Add default class to all buttons */
+    @Input() buttonClass: string = '';
+
```

Comment:
Changes in this file are truly the only ones that  should be kept in this commit, which is about "**Exposing  'buttonClass' input to set custom classes to all buttons**"

Other changes:
- improving folder structure by moving files (pipes, classes, styles)
- refactoring of ShareButtonsService
-...

should be in separate consecutive commits (ideally on **different  git branches**, to ease merging or cherry-picking of commits)